client/lib/invites/actions.js

```diff
@@ -8,46 +8,46 @@ import { get } from 'lodash';
 /**
  * Internal dependencies
  */
-import Dispatcher from 'calypso/dispatcher';
 import wpcom from 'calypso/lib/wp';
-import { action as ActionTypes } from 'calypso/lib/invites/constants';
 import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
 import { errorNotice, successNotice } from 'calypso/state/notices/actions';
 import { acceptedNotice } from 'calypso/my-sites/invites/utils';
 import { requestSites, receiveSites } from 'calypso/state/sites/actions';
 import { requestSiteInvites } from 'calypso/state/invites/actions';
+import { INVITE_ACCEPTED } from 'calypso/state/action-types';
 
 /**
  * Module variables
  */
 const debug = new Debug( 'calypso:invites-actions' );
 
-export function createAccount( userData, invite, callback ) {
-	const send_verification_email = userData.email !== invite.sentTo;
+function inviteAccepted( invite ) {
+	return { type: INVITE_ACCEPTED, invite };
+}
 
+export function createAccount( userData, invite ) {
 	return ( dispatch ) => {
-		wpcom
-			.undocumented()
-			.usersNew(
-				Object.assign( {}, userData, { validate: false, send_verification_email } ),
-				( error, response ) => {
-					const bearerToken = response && response.bearer_token;
-					if ( error ) {
-						if ( error.message ) {
-							dispatch( errorNotice( error.message ) );
-						}
-						recordTracksEvent( 'calypso_invite_account_creation_failed', {
-							error: error.error,
-						} );
-					} else {
-						recordTracksEvent( 'calypso_invite_account_created', {
-							is_p2_site: get( invite, 'site.is_wpforteams_site', false ),
-							inviter_blog_id: get( invite, 'site.ID', false ),
-						} );
-					}
-					callback( error, bearerToken );
+		const result = wpcom.undocumented().usersNew( {
+			...userData,
+			validate: false,
+			send_verification_email: userData.email !== invite.sentTo,
+		} );
+
+		result
+			.then( () => {
+				recordTracksEvent( 'calypso_invite_account_created', {
+					is_p2_site: get( invite, 'site.is_wpforteams_site', false ),
```

Comment:
Optional: wanna refactor some `get()` to optional chaining and null coalescing since you're passing by? 😉

Reply:
I'm not excited about this suggestion 🙂 The PR is already big enough, with 10 refactorings.